### DISCUSSION TITLE
[Question] Will scripted / hand-crafted agents be available?

### DISCUSSION THREAD

**Discussion by ykawamura96** [isaac-sim/IsaacLab#15] [Q&A] [answered]
### Question
Hi, thank you for the great project.
The paper says that the demo in Figure 6 was collected by "Demonstration of the designed tasks using hand-crafted state machines and task-space controllers." but this agent was not found in current code base.  Also, there was no specific mention of a hand crafted agent in the [develeopment roadmap.](https://isaac-orbit.github.io/orbit/source/refs/roadmap.html)
Is this something you plan to release in the future?

Thanks!

**Comment by Mayankm96** [comment, marked as answer]
Hi @ykawamura96 

Thanks a lot for the feedback! Yes, we are aiming to release some hand-crafted state machines in the coming months (tentatively April). Currently, we have these implemented using torch backend but we are considering moving them to [warp](https://github.com/NVIDIA/warp) to have an easier way to define a parallelized version of these.

**Reply by Mayankm96** [reply]
@ykawamura96 I added a warp-based state machine for the lift environment. You can find instructions to run it over here: https://isaac-orbit.github.io/orbit/source/setup/sample.html#state-machine

It would be a lot better to have a more general way to define state machines. But that's still a work-in-progress. Let me know if you have further questions.
